aws/sqs/stats.go

```
@@ -0,0 +1,123 @@
+package sqs
+
+import (
+	"fmt"
+	"time"
+)
+
+type Stats struct {
+	TypeOfCheck string
+	Critical    map[string]string
+	Warning     map[string]string
+}
+
+func NewStats(typeOfCheck string) *Stats {
+	return &Stats{
+		typeOfCheck,
+		make(map[string]string),
+		make(map[string]string),
+	}
+}
+
+func (s *Stats) Generate(cfg *Configuration, queues []*QueueInfo) {
+	if cfg.TypeOfCheck == "AgeOfMessages" {
+		for _, q := range queues {
+			if q.CheckEnvironment(cfg.Environment) && q.CheckTeam(cfg.Team) {
+				s.classifyAgeOfOldestMessagesInQueue(q, cfg)
+			}
+		}
+	}
+	if cfg.TypeOfCheck == "NumberOfMessages" {
+		for _, q := range queues {
+			if q.CheckEnvironment(cfg.Environment) && q.CheckTeam(cfg.Team) {
+				s.classifyNumberOfMessagesInQueue(q, cfg)
```

same here, ignored errors

---

fixed here: https://github.com/AirHelp/sensu-plugins-go/pull/11/commits/b4c8960f7a183e322c108ec5056a94f7de1f133b#diff-0374bb1d8da8defe98a02c980a08b7feR79